### DISCUSSION TITLE
[mania + tablet] Limit to 4K on tablet

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [ppy/osu#18658] [Ideas] [answered]
It is known that mania is kinda playable on lazer out of the box on tablet, but virtually all general purpose multi-touch device (that includes your daily tablet/phone/POS machine/etc) in existence only supports up to 4 fingers input and there are often no touch event exceeding that maximum. This implies 5K and above are also virtually impossible to play properly, especially when there are full drops (all keys press). This would often lead to game pause, most likely due to accidentally triggering touch gesture to move apps. This phenomena is very rampant on other mobile rhythm game and devices (cough cough Huawei) as well.

**Comment by bdach** [comment, marked as answer]
> virtually all general purpose multi-touch device (that includes your daily tablet/phone/POS machine/etc) in existence only supports up to 4 fingers input

[citation needed]

Both of my android test devices support at least 10 simultaneous touches. One of them is at least 5 years old now. Android API puts no theoretical limit to the number of simultaneous touches, and even iOS supports up to 10. Also you can always plug in an alternate input device like an external keyboard..

In other words, I find no basis to limit to 4k on any device, and even less basis to try and detect if one can support it. If it doesn't work due to the device's limitation, tough luck.
